hibike/devices/YogiBear/ui.cpp

```diff
@@ -0,0 +1,173 @@
+    }
+    else if (c == 'r') {
+      Serial.print("Encoder Pos: ");
+      Serial.println(readPos());
+      Serial.print("Encoder Vel: ");
+      Serial.println(readVel());
+      Serial.print("Current: ");
+      Serial.println(readCurrent());
+    }
+    else if (c == 't') {
+      continualPrint = !continualPrint;
+      Serial.println("Toggling readout prints");
+    }
+    else if (c == 'w') {
+      Serial.read();
+      char val1 = Serial.read();
+      char val2 = Serial.read();
+      
+      if (val1 == 'c') {
+        setCurrentThreshold(Serial.parseFloat());
+        Serial.print("New current threshold: ");
+        Serial.println(current_threshold);
+      }
+      else if (val1 == 'p') {
+        if (val2 == 'p'){
+          setPosKP(Serial.parseFloat());
+          Serial.print("New PID position KP value: ");
+          Serial.println(PIDPosKP);
+        }
+        else if (val2 == 'i'){
+          setPosKI(Serial.parseFloat());
+          Serial.print("New PID position KI value: ");
+          Serial.println(PIDPosKI);
+        }
+        else if (val2 == 'd'){
+          setPosKD(Serial.parseFloat());
+          Serial.print("New PID position KD value: ");
+          Serial.println(PIDPosKD);
+        }
+        else {
+          setPosSetpoint(Serial.parseFloat());
+          Serial.print("New PID position: ");
+          Serial.println(PIDPos);
+        }
+        updatePosPID();
+
+    }
+    else if (val1 == 'v') {
+      if (val2 == 'p') {
+        setVelKP(Serial.parseFloat());
+        Serial.print("New PID velocity KP value: ");
+        Serial.println(PIDVelKP);
+      }
+      else if (val2 == 'i') {
+        setVelKI(Serial.parseFloat());
+        Serial.print("New PID velocity KI value: ");
+        Serial.println(PIDVelKI);
+      }
+      else if (val2 == 'd') {
+        setVelKD(Serial.parseFloat());
+        Serial.print("New PID velocity KD value: ");
+        Serial.println(PIDVelKD);
+      } 
+      else {
+        setVelSetpoint(Serial.parseFloat());
+        Serial.print("New PID velocity: ");
+        Serial.println(PIDVel);
+      }
+      updateVelPID();
+    }
+    
+    else {
+      Serial.println("Cannot write to that");
+    }
+  }
+  else if (c == 'b') {
+    Serial.println("Heartbeat");
+    heartbeat = 0;
+  }
+  else if (c == 'h') {
+    hibike = true;
+    Serial.println("Going to hibike mode");
+  }
+  else if (c == 'z') {
+    hibike = false;
+    Serial.println("Turning off hibike");
+  }
+  else if (c == '?') {
+    Serial.println("Manual Controls: \ns <x> - sets pwm to x \nc     - clears faults \np <x> - turns on PID mode, velocity if x = v, position if x = p \nm     - turns on manual input mode \ne     - enables motor \nd     - disables motor \nr     - displays 1 print of all readable values \nt     - toggles continual printing of pos and vel \nb     - send heartbeat \nh     - switch hibike mode \nz     - switch human controls \nw <x> <y> - writes the value y to the variable x");
```

Comment:
FYI there are better approaches then having this entire string on one line.

Example: http://stackoverflow.com/questions/797318/how-to-split-a-string-literal-across-multiple-lines-in-c-objective-c